src/lib/features/personal-dashboard/personal-dashboard-service.ts

```diff
@@ -105,6 +111,12 @@ export class PersonalDashboardService {
         userId: number,
         projectId: string,
     ): Promise<PersonalDashboardProjectDetailsSchema> {
+        if (!(await this.projectStore.hasProject(projectId))) {
```

Comment:
I'm not sure about introducing projectStore since it brings coupling to the entire API of this store. What if we could call `getOnboardingStatusForProject(projectId: string)` before the whole Promise.all and let it return null when the status doesn't exist which essentially translates to 404. This will also make `getOnboardingStatusForProject` more resilient.